accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,530 @@
+
+As such, what we care about in this threat model is whether this provenance information can be forged and whether verification can be bypassed.
+
+The following table highlights the impact of different compromises:
+
+| Compromised infrastructure | Forge provenance                 | Bypass verification |
+|----------------------------|----------------------------------|--------------------|
+| npm registry               | No                               | Yes                |
+| CI/CD                      | Yes                              | No                 |
+| Sigstore                   | Yes                              | No                 |
+| Network (MITM)             | No                               | Yes                |
+| User account (npm)         | No                               | Yes                |
+| User account (GH)          | No                               | No                 |
+| Trusted builder            | Likely                           | No                 |
+
+An attacker could forge what goes in the provenance information if the CI/CD provider or Sigstore was compromised. Similarly, if the trusted builder was compromised in some way it's likely the attacker could forge the provenance information during the build. Mitigating these attacks is out of scope in this proposal.
+
+An attacker could bypass verification if the npm registry, npm user or the network was compromised by not providing provenance information or removing it in transit.
+
+The long-term solution to bypassing verification is enforcing or requiring that all packages have provenance information set during install time in the npm CLI. It might take years to get to the point where a large portion of npm packages have this information set so we'll need to find ways to get there gradually.
+
+We could make it harder to remove provenance information once it's been set by a maintainer. For example, blocking a npm publish that doesn't include it if the current latest version does include it. Requiring user 2FA to remove it once it's set.
+
+Maintainers and developers consuming packages could also opt-in to requiring all their dependencies to include provenance information once their dependencies include this.
+
+## Rationale and Alternatives
+While no existing solution addresses all of the goals outlined above at this time, the [Sigstore](https://www.sigstore.dev/) project and its use of short-lived X.509 certificates is closest to our stated goals for the vast majority of developers since it reduces the complexity of key management, and removes the need for revocation scenarios.
+
+Using OIDC to handle authentication of workloads requesting a short-lived X.509 certificate allows this solution to be vendor neutral and work consistently across many source repositories and CI systems.
+
+
+### Signing methods
```

Comment:
Another alternatives is Long-lived CI/CD managed keys. For example, I believe Google Cloud Build signs all of its provenance attestations with some fixed public key. I believe this would model still address most or all of the requirements. Under the current proposal, NPM already has to maintain a list of trusted builders; using a fixed key per builder would just add one more piece of data to that list. Maintenance might be a bit more difficult, but I think it's worth adding to this list.

Reply:
> Another alternatives is Long-lived CI/CD managed keys

Yes! We actually floated this idea early on but ran into questions around key revocation and distribution. Will add this as a alternative 👍 

I get the feeling GCP is minting one key-pair per project or repository from [glancing the docs](https://cloud.google.com/build/docs/securing-builds/view-build-provenance) but might be wrong?

Reply:
Oops, I think you're right. Let's just remove GCB as an example and speak hypothetically.